crates/macro/src/lib.rs

```diff
@@ -18,6 +18,26 @@ pub fn wasm_bindgen(attr: TokenStream, input: TokenStream) -> TokenStream {
     }
 }
 
+/// This macro adds a linked module by module, raw_module or inline_js attribute.
+/// It expands to a String containing a URL to that module. This URL can be used
+/// to create workers or worklets, for example:
+/// ```no_run
+/// use web_sys::Worker;
+/// let worker = Worker::new(&wasm_bindgen::link_to!(module = "/src/worker.js"));
+/// ```
+#[proc_macro]
+pub fn link_to(input: TokenStream) -> TokenStream {
+    match wasm_bindgen_macro_support::expand_link_to(input.into()) {
+        Ok(tokens) => {
+            if cfg!(feature = "xxx_debug_only_print_generated_code") {
+                println!("{}", tokens);
+            }
+            tokens.into()
+        }
+        Err(diagnostic) => (quote! { String::clone(#diagnostic) }).into(),
```

Comment:
Why is this `String::clone` here?

Reply:
The invalid expression should have the type `String` so the compiler or the analyzer can still typecheck the remaining program, even if the arguments are invalid.

Reply:
You should probably add a comment about it, though:

```suggestion
        // This `String::clone` is here so that IDEs know this is supposed to be a
        // `String` and can keep type-checking the rest of the program even if the macro
        // fails.
        Err(diagnostic) => (quote! { String::clone(#diagnostic) }).into(),
```